### PR TITLE
patch chart and values

### PR DESCRIPTION
**Pull request by anvddriesch**
This PR:

- changes vendir to sync the whole chart and not just the template (that's why binary and so on is now here)
- adds my first clumsy attempt at patching the chart. I just created the diff between the upstream `values.yaml` and `Chart.yaml` to ours and use them as a patch
- the sync.sh script is based on this here https://github.com/giantswarm/cilium-app/blob/main/sync/sync.sh, running it syncs vendir and then applies patches

### Testing

Description on how csi-driver-nfs can be tested.

- [ ] fresh install works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM
- [ ] upgrade from previous version works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM

#### Other testing

Description of features to additionally test for csi-driver-nfs installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: sync/patches/values/_values.yaml.patch [hunk 1]
```diff
@@ -0,0 +1,103 @@
+diff --git b/helm/csi-driver-nfs/values.yaml a/helm/csi-driver-nfs/values.yaml
```
Comment (anvddriesch) [inline sync/patches/values/_values.yaml.patch:1]: these diffs are really ugly and I'm thinking we need to restructure them if we want to have a more readable patch

File: vendir.yml [hunk 2]
```diff
@@ -10,9 +10,9 @@ directories:
           ref: "v4.9.0"
         includePaths:
           - charts/latest/**/*
-  - path: helm/csi-driver-nfs/templates
+  - path: helm
     contents:
     - path: .
       directory:
-        path: vendor/csi-driver-nfs/charts/latest/csi-driver-nfs/templates
+        path: vendor/csi-driver-nfs/charts/latest
```
Comment (anvddriesch) [inline vendir.yml:17]: we now sync the whole chart

File: helm/csi-driver-nfs/.helmignore [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Patterns to ignore when building packages.
```
Comment (anvddriesch) [inline helm/csi-driver-nfs/.helmignore:1]: this little guy just appeared during sync

**Review by vxav — APPROVED** [review]
Looks alright to me

**Pull request merged**